src/main/java/parser/CommandParser.java

```diff
@@ -0,0 +1,85 @@
+package parser;
+
+import java.util.Scanner;
+
+import command.Command;
+import command.CommandType;
+import task.TaskType;
+import time.Time;
+
+/*
+ * Singleton class
+ * parse commandType, Time, TaskDescription
+ */
+public class CommandParser {
+    private static final Scanner scan = new Scanner(System.in);
+    private static CommandParser commandParser = null;
+
+    /**
+     * Constructor
+     */
+    private CommandParser() {}
+
+    /**
+     * Singleton return
+     */
+    public static CommandParser getCommandParser() {
+        if (commandParser == null) {
+            commandParser = new CommandParser();
+        }
+        return commandParser;
+    }
+
+
+    /**
+     * Parses input one line at a time and return a command object
+     */
+    public Command parseNextCommand() {
+
+        String cmdStr = scan.nextLine();
+        String[] tokens = cmdStr.toLowerCase().split(" ");
+        Command cmd = new Command();
+
+        for (int i = 0; i < tokens.length; i++) {
+            String token = tokens[i];
+            TokenType tokenType = TokenType.getTokenTypebyStr(token);
+            if (tokenType == TokenType.TASK_TYPE) {
+                cmd.setTaskType(parseTaskType(token));
+            } else if (tokenType == TokenType.COMMAND_TYPE) {
+                // TODO: Add handling to not set command types twice
+                cmd.setCommandType(parseCommandType(token));
+            } else if (tokenType == TokenType.TIME_TYPE) {
+                // TODO: Assumption here is that after time token, the rest of
+                // the string is all related to time
+                String timeToken = new String("");
+                for (int j = i; j < tokens.length; j++) {
+                    timeToken += (tokens[j] + " ");
+                }
+                cmd.setTimeInfo(parseTimeInfo(timeToken));
+                break; // end of parsing
+            } else if (tokenType == TokenType.DESCRIPTION) {
+                cmd.appendTaskDescription(token + " ");
+            } else {
+                System.out.println("[PARSER] Invalid token found from input! Token: " + token);
```

Comment:
consider putting this logic in a separate function to improve readability of your code and make your methods more modular and shorter than 30 LoCs (consider 30 as some sort of a rule to follow)